044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,345 @@
+|----------------------------------------------------------------|--------|---------|-------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| client.quota.callback.static.storage.volume.source             | string | local   | local, cluster    | set to cluster to obtain volume descriptions from cluster, see [VolumeSource](#volume-source)                                    |
+| client.quota.callback.static.storage.check-interval            | long   | 0       | (0, ...)          | 0 means disabled, otherwise this is the milliseconds between polling to describe the cluster (this is an existing configuration) |
+| client.quota.callback.static.kafka.admin.bootstrap.servers     | string | null    |                   | required if volume source is cluster, bootstrap.servers for [admin client](#admin-client-configuration) used to get cluster data |
+| client.quota.callback.static.kafka.admin.*                     | ?      |         |                   | optionally users can configure arbitrary properties of the [admin client config](#admin-client-configuration)                    |
+| client.quota.callback.static.throttle.factor.fallback          | double | 1.0     | (0.0, 1.0)        | sets the [Throttle Factor Fallback](#throttle-factor-fallback)                                                                   |
+| client.quota.callback.static.throttle.factor.validity-duration | string | PT5M    | ISO8601 durations | sets the [Throttle Factor Validity Duration](#throttle-factor-validity-duration)                                                 |
+
+### Limit Configuration
+When using cluster sourced volumes the user must configure a single [limit type](#limit-type-configuration), all are incompatible with `local` volume source
+
+|                                                                         | type   | default | valid values |                                                         |
+|-------------------------------------------------------------------------|--------|---------|--------------|---------------------------------------------------------|
+| client.quota.callback.static.storage.perVolumeLimit.availableBytesBelow | long   | null    | (0, ...)     | stop message production if availableBytes <= this value |
+| client.quota.callback.static.storage.perVolumeLimit.availableRatioBelow | double | null    | (0.0, 1.0)   | stop message production if availableRatio <= this value |
+
+## Metrics
+
+1. Add a `io.strimzi.kafka.quotas:type=LocalThrottleFactor,name=ThrottleFactor` gauge. This would emit the most recently calculated throttle factor.
+2. Add a `io.strimzi.kafka.quotas:type=LocalThrottleFactor,name=FallbackThrottleFactorApplied` counter.
+3. Add a `io.strimzi.kafka.quotas:type=LocalThrottleFactor,name=ThrottlingVolume,brokerId=1,volumeLogDir=/path` gauge. The volume that most recently and severely breached a limit.
```

Comment:
maybe change `brokerId=1` to `brokerId=([0-9]+)`?

Reply:
We've realized this gauge would suffer from the same issue as the other metric we changed (the gauge would live on forever after a single failure and would be misleading).

Alternatively we could change to a monotonic counter like `LimitViolation` and rely on logs to find the volume

Reply:
pushed up a change to counter